domains/auth/Auth.API/Tests.Integration/Controllers/TermsControllerTests.cs

```diff
@@ -0,0 +1,145 @@
+
+        var dto = new AcceptTermsDTO(2);
+        var httpContent = new StringContent(JsonConvert.SerializeObject(dto), Encoding.UTF8, "application/json");
+        var result = await client.PutAsync("terms/accept", httpContent);
+
+        var dbUser = factory.ServiceProvider.GetRequiredService<DataContext>().Users.AsNoTracking().SingleOrDefault(x => x.Id == user.Id)!;
+
+        Assert.NotNull(result);
+        Assert.Equal(HttpStatusCode.NoContent, result.StatusCode);
+        Assert.Equal(user.ProviderId, dbUser.ProviderId);
+        Assert.Equal(user.Name, dbUser.Name);
+        Assert.Equal(user.Tin, dbUser.Tin);
+        Assert.Equal(user.Id, dbUser.Id);
+        Assert.Equal(user.AllowCPRLookup, dbUser.AllowCPRLookup);
+        Assert.Equal(dto.Version, dbUser.AcceptedTermsVersion);
+    }
+
+    [Fact]
+    public async Task AcceptTermsAsync_ShouldReturnNoContentAndCreateUser_WhenUserDoesNotExist()
+    {
+        var user = await factory.AddUserToDatabaseAsync();
+
+        var name = Guid.NewGuid().ToString();
+        var providerId = Guid.NewGuid().ToString();
+        var tin = null! as string;
+        var allowCprLookup = false;
+        var acceptedTermsVersion = 0;
+
+        var client = await factory.CreateAuthenticatedClientAsync(user, config: builder =>
+        {
+            var mapper = Mock.Of<IUserDescriptMapper>();
+            _ = Mock.Get(mapper)
+                .Setup(x => x.Map(It.IsAny<ClaimsPrincipal>()))
+                .Returns(value: new UserDescriptor(null!)
+                {
+                    Id = null,
+                    Name = name,
+                    ProviderId = providerId,
+                    Tin = tin,
+                    AllowCPRLookup = allowCprLookup,
+                    AcceptedTermsVersion = acceptedTermsVersion
+                });
+
+            builder.ConfigureTestServices(services =>
+            {
+                services.AddScoped(x => mapper);
+            });
+        });
+
+        var dto = new AcceptTermsDTO(1);
+        var httpContent = new StringContent(JsonConvert.SerializeObject(dto), Encoding.UTF8, "application/json");
+        var result = await client.PutAsync("terms/accept", httpContent);
+
+        var dbUser = factory.ServiceProvider.GetRequiredService<DataContext>().Users.AsNoTracking().SingleOrDefault(x => x.ProviderId == providerId)!;
+
+        Assert.NotNull(result);
+        Assert.Equal(HttpStatusCode.NoContent, result.StatusCode);
+        Assert.Equal(providerId, dbUser.ProviderId);
+        Assert.Equal(name, dbUser.Name);
+        Assert.Equal(tin, dbUser.Tin);
+        Assert.Equal(allowCprLookup, dbUser.AllowCPRLookup);
+        Assert.Equal(dto.Version, dbUser.AcceptedTermsVersion);
+    }
+
+    [Fact]
+    public async Task AcceptTermsAsync_ShouldReturnInternalServerError_WhenUserDescriptMapperReturnsNull()
+    {
+        var user = await factory.AddUserToDatabaseAsync();
```

Comment:
Try with just changing the `Id` after creating the user in the database and then not mocking so much.